### PR TITLE
Multiple commits with UI fixes, see description

### PR DESCRIPTION
**Pull request by Karakatiza666**
Fix https://github.com/feldera/feldera/issues/854 : Cannot delete a misconfigured output connector

Fix https://github.com/feldera/feldera/issues/935 : Replace youtube link on Webconsole UI home page

Fix https://github.com/feldera/feldera/issues/963 : Kafka authentication protocol security.protocol field

Fix https://github.com/feldera/feldera/issues/695 : Differentiate compiling and pending in web-console editor

Fix https://github.com/feldera/feldera/issues/776 : Report number of parsing errors per connector.

Fix https://github.com/feldera/feldera/issues/635 : Table browser doesn't always correctly apply updates - no longer reproduces

Fix https://github.com/feldera/feldera/issues/494 : MUI DataGridPro fails to load data on Next.js > 13.4.8-canary.9 - no longer reproduces

Fix https://github.com/feldera/feldera/issues/932 : Spacing for SQL icon is off in pipeline view

Also includes numerous refactorings

==============================

Fix https://github.com/feldera/feldera/issues/1012 : doesn't reproduce with this PR

### CODE REVIEW

**Comment by gz** [comment]
image

trying to connect two views to an output gives an error (entity with this name already exists)

**Comment by gz** [comment]
these boxes look odd because they aren't sized the same for the same number of digits:

image

**Comment by gz** [comment]
different font size for `connector errors:` and number is odd also it looks like only botton corners are rounded but not the top IMO would look better if either all corners or none are rounded

image

**Comment by gz** [comment]
most of the time the tput curve just shows a straight line .. would be good to have slightly more history in the curve (e.g., the pattern is periodic but it doesn't reflect that)

https://github.com/feldera/feldera/assets/127654/fbbc2a57-1877-4c7a-9c9c-a61cdacfdf38

**Comment by gz** [comment]
editing a connector causes browser crash:

https://github.com/feldera/feldera/assets/127654/d70bb867-e0b7-4367-962f-dfa635eb8de6

**Comment by gz** [comment]
I think #865 isn't fully fixed because the editor still makes some 3rd party request e.g. if I open the SQL editor or generic connector editor dialog this is what I'm seeing:

image

File: web-console/src/app/(root)/(authenticated)/(app)/streaming/builder/page.tsx [hunk 1]
```diff
@@ -111,11 +115,11 @@ const PipelineWithProvider = (props: {
       return
     }
     const isReady = !(
-      pipelineQuery.isLoading ||
+      pipelineQuery.isPending ||
```
[the remark this answers is not in the document]
Comment (Karakatiza666) [reply web-console/src/app/(root)/(authenticated)/(app)/streaming/builder/page.tsx:118]: Yes

**Comment by Karakatiza666** [comment]
Updated design for Active Pipeline on Home page: 
image
image

**Comment by Karakatiza666** [comment]
Made the color to take up the entire width of a column
image

File: web-console/src/lib/functions/common/tanstack.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+import { InvalidateQueryFilters, QueryClient, Updater, UseQueryOptions } from '@tanstack/react-query'
 
 type FunctionType = (...args: any) => any
 type Arguments<F extends FunctionType> = F extends (...args: infer A) => any ? A : never
 
 type QueryType<U extends Record<string, FunctionType>, P extends keyof U> = {
   queryKey: readonly unknown[]
   queryFn: () => ReturnType<U[P]>
-} & Pick<UseQueryOptions<ReturnType<U[P]>, ApiError, Awaited<ReturnType<U[P]>>, readonly unknown[]>, 'onError'>
+} & UseQueryOptions<Awaited<ReturnType<U[P]>>, ApiError, Awaited<ReturnType<U[P]>>, readonly unknown[]> // Pick<UseQueryOptions<ReturnType<U[P]>, ApiError, Awaited<ReturnType<U[P]>>, readonly unknown[]>, 'onError'>
```
[the remark this answers is not in the document]
Comment (Karakatiza666) [reply web-console/src/lib/functions/common/tanstack.ts:11]: This file is permanent and simplifies working with Tanstack Query; added a few explanatory comments to this file

**Comment by Karakatiza666** [comment]
> trying to connect two views to an output gives an error (entity with this name already exists)
This behavior is on latest `main`, too
Let me know what is the intended behavior (connection should get created, or second connection to an output connector should not be allowed) and I'll create an issue for that

**Review by gz — APPROVED** [review]

**Comment by gz** [comment]
> Let me know what is the intended behavior

should be allowed AFAICT

**Pull request merged**